packages/swagger-ui/src/index.ts

```diff
@@ -0,0 +1,53 @@
+import type { Env, MiddlewareHandler } from 'hono'
+import { html } from 'hono/html'
+import { remoteAssets } from './swagger/resource'
+
+type ResourceOptions = {
+  version?: string
+}
+
+type SwaggerUIOptions = {
+  url: string
+  ui?: ResourceOptions
+}
+
+const SwaggerUI = (options: SwaggerUIOptions) => {
+  const asset = remoteAssets({ version: options?.ui?.version })
+
+  return html`
+    <div>
+      <div id="swagger-ui"></div>
+      ${asset.css.map((url) => html`<link rel="stylesheet" href="${url}" />`)}
+      ${asset.js.map((url) => html`<script src="${url}" crossorigin="anonymous"></script>`)}
+      <script>
+        window.onload = () => {
+          window.ui = SwaggerUIBundle({
```

Comment:
What if we could pass options to the `SwaggerUIBundle`? For example, a user might want to customize like this:

```js
    window.onload = () => {
      window.ui = SwaggerUIBundle({
        url: '${options.url}',
        dom_id: '#swagger-ui',
        presets: [
          SwaggerUIBundle.presets.apis,
          SwaggerUIStandalonePreset
        ],
        layout: "StandaloneLayout",
      });
    };
```

You can refer to https://swagger.io/docs/open-source-tools/swagger-ui/usage/installation/.

Reply:
thx!
I'll give it a try!

Reply:
Hi, @yusukebe @naporin0624

I tried some and I figured out that `SwaggerUIStandalonePreset`, `SwaggerUIBundle.presets.apis`, etc… is a function.

I think it’s difficult to apply the configuration object to string html.

Do you have any plan or thought for this?

Reply:
I suppose it could be configured as is with a string, but I believe there is a DX-like problem.

Reply:
Hi @sor4chi,

I thought these should be strings like this:

```ts
type SwaggerUIOptions = {
  url: string;
  ui?: ResourceOptions;
  presets?: string[];
  layout?: string;
}
```

And the implementation is like this:

```ts
const optionsStrings = Object.entries(options)
  .map(([k, v]) => {
    return `${k}: '${Array.isArray(v) ? v.join(',') : v}'`;
  })
  .join(',\n');
```

> I believe there is a DX-like problem.

I think it's OK for them to be strings because the most important thing is that users can set their own settings.

Reply:
https://www.npmjs.com/package/@types/swagger-ui-dist?activeTab=code
Are you planning to not support other options?

Reply:
@sor4chi 

You found it well. I think it's better to support it.

Reply:
I tried it out, but there is actually a place to write a callback function in the options of this library.
I think it would be difficult to fully support it, and some different approach may be needed.

Reply:
@sor4chi

I think that `operationsSorter`, `requestInterceptor`, and `responseInterceptor` can be strings. They may not be used often, so it's not a big problem to write the functions as strings.

Reply:
alright! thank you for your opinion!